test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -0,0 +1,284 @@
+    private static final SubpopulationGuid SUBPOP_GUID = SubpopulationGuid.create("ABC");
+    private static final ConsentStatus CONSENTED_STATUS = new ConsentStatus.Builder().withName("Name")
+            .withGuid(SUBPOP_GUID).withRequired(true).withConsented(true).build();
+    private static final ConsentStatus UNCONSENTED_STATUS = new ConsentStatus.Builder().withName("Name")
+            .withGuid(SUBPOP_GUID).withRequired(true).withConsented(false).build();
+    private static final Map<SubpopulationGuid, ConsentStatus> CONSENTED_STATUS_MAP = new ImmutableMap.Builder<SubpopulationGuid, ConsentStatus>()
+            .put(SUBPOP_GUID, CONSENTED_STATUS).build();
+    private static final Map<SubpopulationGuid, ConsentStatus> UNCONSENTED_STATUS_MAP = new ImmutableMap.Builder<SubpopulationGuid, ConsentStatus>()
+            .put(SUBPOP_GUID, UNCONSENTED_STATUS).build();
+    private static final CriteriaContext CONTEXT = new CriteriaContext.Builder()
+            .withStudyIdentifier(TestConstants.TEST_STUDY).build();  
+    private static final StudyParticipant PARTICIPANT = new StudyParticipant.Builder().build();
+
+    @Mock
+    private CacheProvider cacheProvider;
+    @Mock
+    private BridgeConfig config;
+    @Mock
+    private ConsentService consentService;
+    @Mock
+    private ParticipantOptionsService optionsService;
+    @Mock
+    private AccountDao accountDao;
+    @Mock
+    private ParticipantService participantService;
+    @Mock
+    private SendMailService sendMailService;
+    @Mock
+    private StudyService studyService;
+    @Mock
+    private PasswordResetValidator passwordResetValidator;
+    @Mock
+    private Study study;
+    @Mock
+    private Account account;
+    @Mock
+    private UserSession session;
+    
+    @Captor
+    private ArgumentCaptor<String> stringCaptor;
+    @Captor
+    private ArgumentCaptor<EmailSignInEmailProvider> providerCaptor;
+    
+    @Spy
+    private AuthenticationService service;
+
+    @Before
+    public void before() {
+        service.setCacheProvider(cacheProvider);
+        service.setBridgeConfig(config);
+        service.setConsentService(consentService);
+        service.setOptionsService(optionsService);
+        service.setAccountDao(accountDao);
+        service.setPasswordResetValidator(passwordResetValidator);
+        service.setParticipantService(participantService);
+        service.setSendMailService(sendMailService);
+        service.setStudyService(studyService);
+
+        doReturn(study).when(studyService).getStudy(STUDY_ID);
+        doReturn(STUDY_ID).when(study).getIdentifier();
+        doReturn(true).when(study).isEmailSignInEnabled();
+        doReturn(new StudyIdentifierImpl(STUDY_ID)).when(study).getStudyIdentifier();
+        doReturn(new EmailTemplate("subject","body",MimeType.TEXT)).when(study).getEmailSignInTemplate();
+        doReturn("sender@sender.com").when(study).getSupportEmail();
+        doReturn("Sender").when(study).getName();
```

Comment:
Study is a data type with no behaviors that need mocking, and we shouldn't be changing this anytime soon. Would it be better to create a concrete Study (using Study.create()) and just filling in the actual values instead of using a mock?